webhook/deployment/deployment.go

```diff
@@ -0,0 +1,350 @@
+	return err
+}
+
+func removeServiceIfExists(serviceName string) {
+	service, err := clientset.CoreV1().Services(namespace).Get(serviceName, metav1.GetOptions{})
+	if service != nil && err == nil {
+		logging.Debugf("service %s already exists, removing it first", serviceName)
+		err := clientset.CoreV1().Services(namespace).Delete(serviceName, &metav1.DeleteOptions{})
+		if err != nil {
+			fmt.Errorf("error trying to remove service: %s", err)
+		}
+		logging.Debugf("service %s removed", serviceName)
+	}
+}
+
+func removeValidatingWebhookIfExists(configName string) {
+	validatingWebhok, err := clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Get(configName, metav1.GetOptions{})
+	if validatingWebhok != nil && err == nil {
+		logging.Debugf("validating webhook %s already exists, removing it first", configName)
+		err := clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Delete(configName, &metav1.DeleteOptions{})
+		if err != nil {
+			fmt.Errorf("error trying to remove validating webhook configuration: %s", err)
+		}
+		logging.Debugf("validating webhook configuration %s removed", configName)
+	}
+}
+
+func removeMutatingWebhookIfExists(configName string) {
+	config, err := clientset.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Get(configName, metav1.GetOptions{})
+	if config != nil && err == nil {
+		logging.Debugf("mutating webhook %s already exists, removing it first", configName)
+		err := clientset.AdmissionregistrationV1beta1().MutatingWebhookConfigurations().Delete(configName, &metav1.DeleteOptions{})
+		if err != nil {
+			fmt.Errorf("error trying to remove mutating webhook configuration: %s", err)
+		}
+		logging.Debugf("mutating webhook configuration %s removed", configName)
+	}
+}
+
+func removeSecretIfExists(secretName string) {
+	secret, err := clientset.CoreV1().Secrets(namespace).Get(secretName, metav1.GetOptions{})
+	if secret != nil && err == nil {
+		logging.Debugf("secret %s already exists, removing it first", secretName)
+		err := clientset.CoreV1().Secrets(namespace).Delete(secretName, &metav1.DeleteOptions{})
+		if err != nil {
+			fmt.Errorf("error trying to remove secret: %s", err)
+		}
+		logging.Debugf("secret %s removed", secretName)
+	}
+}
+
+func deploy() {
+	/* generate CSR and private key */
+	csr, key, err := generateCSR()
+	if err != nil {
+		logging.Errorf("error generating CSR and private key: %s", err)
+		os.Exit(1)
+	}
+	logging.Debugf("raw CSR and private key succesfully created")
+
+	/* obtain signed certificate */
+	certificate, err := getSignedCertificate(csr)
+	if err != nil {
+		logging.Errorf("error getting signed certificate: %s", err)
+		os.Exit(1)
+	}
+	logging.Debugf("signed certificate succesfully obtained")
+
+	/* create secret and push it to the API */
+	err = createSecret(certificate, key)
+	if err != nil {
+		logging.Errorf("error creating secret: %s", err)
+		os.Exit(1)
+	}
+	logging.Debugf("secret succesfully created")
```

Comment:
nit: successfully (two s), here and below

Reply:
fixed